lib/semantics/check-call.cc

```diff
@@ -437,6 +450,105 @@ static void CheckExplicitDataArg(const characteristics::DummyDataObject &dummy,
   }
 }
 
+static void CheckProcedureArg(evaluate::ActualArgument &arg,
+    const characteristics::DummyProcedure &proc, const std::string &dummyName,
+    evaluate::FoldingContext &context) {
+  parser::ContextualMessages &messages{context.messages()};
+  const characteristics::Procedure &interface{proc.procedure.value()};
+  if (const auto *expr{arg.UnwrapExpr()}) {
+    bool dummyIsPointer{
+        proc.attrs.test(characteristics::DummyProcedure::Attr::Pointer)};
+    const auto *argProcDesignator{
+        std::get_if<evaluate::ProcedureDesignator>(&expr->u)};
+    const auto *argProcSymbol{
+        argProcDesignator ? argProcDesignator->GetSymbol() : nullptr};
+    if (auto argChars{characteristics::DummyArgument::FromActual(
+            "actual argument", *expr, context)}) {
+      if (auto *argProc{
+              std::get_if<characteristics::DummyProcedure>(&argChars->u)}) {
+        characteristics::Procedure &argInterface{argProc->procedure.value()};
+        if (argInterface.attrs.test(
+                characteristics::Procedure::Attr::NullPointer)) {
+        }
```

Comment:
Is something supposed to happen in this `if`?

Reply:
Not any more.